### PR TITLE
Add 4 new values to sales REST endpoint

### PR DESCRIPTION
**Pull request by fpcorso**
Building out a data pipeline for extracting and combining data from our store, support inbox, email list, other plugins/EDD extensions, and others to build out reports. However, we need some additional data points in the sales endpoint.

Proposed Changes:
1. To be able to ensure correct joins with data from other plugins, we need user ID.
2. To ensure we are only building reports on our live data, we need the mode.
3. To combine data with other EDD reports, it would be useful to have customer ID.
4. We will also be adding in other payments with different statuses through some of the filters, such as from the Recurring Payments extension, so we need to be able to see Complete vs Renewal statuses.

Here they are in the response after adding these fields on a local test site:
example-edd-sales

_Please do not submit PRs with minified CSS or JS files. This is managed at the time of release by the Core Team_

### CODE REVIEW

**Comment by fpcorso** [comment]
All three of the failed tests in Travis CI are due to a composer install error.

File: includes/api/class-edd-api-v2.php [hunk 1]
```diff
@@ -389,13 +389,17 @@ public function get_recent_sales() {
 				$user_info    = $payment->user_info;
 
 				$sales['sales'][ $i ]['ID']             = $payment->number;
+				$sales['sales'][ $i ]['mode']           = $payment->mode;
+				$sales['sales'][ $i ]['status']         = $payment->status_nicename;
```
Comment (ashleyfae) [inline includes/api/class-edd-api-v2.php:393]: I think this would be better as `$payment->status`

Comment (fpcorso) [reply includes/api/class-edd-api-v2.php:393]: @ashleyfae, I went back-and-forth on which to do here. If someone was using the REST API with another integration just to quickly generate a report for stakeholders, not sure if "publish" or "edd_subscription" would mean as much as "Complete" or "Renewal". 

But, on the other side, most people who are interacting with the endpoint are probably developers/analysts who could filter/modify the status, if needed.

Yeah, probably better to just go with "status" then.

Comment (fpcorso) [reply includes/api/class-edd-api-v2.php]: Changed.

**Comment by cklosowski** [comment]
Just updating the base branch for this to `release/2.9.24`, we just practice release branching so we can avoid possibly conflicts in `master` :+1:

**Comment by fpcorso** [comment]
@cklosowski No worries. EDD's [contributing guidelines](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/CONTRIBUTING.md#making-changes) say to set the pull request to `master`. For in the future, should I still do that since I wouldn't know which is the current release branch or is there a better branch I should initialize the PR's on? Thanks!

**Comment by cklosowski** [comment]
@fpcorso it's perfectly fine to just go against master, so long as your fork is reasonably up to date, it should have no issues directly merging in. If we need another base for your PR to be submitted we can let you know.

Appreciate the PR!

File: includes/api/class-edd-api-v2.php (hunk 1 above)
Comment (ashleyfae) [reply includes/api/class-edd-api-v2.php]: Thanks for the change!

`status_nicename` is really the "display" value, whereas "status" is the internal one. I think the internal one is better for the API and on the other side people you can convert it to your desired display.

**Pull request merged**